cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        String[] partitionArray = sourcePartitionColumns.split(",");
+        for(int i=0;i<partitionArray.length;i++){
+            partitionDataTypeMap.put(partitionArray[i].toUpperCase().substring(0,partitionArray[i].indexOf(" ")),partitionArray[i].toUpperCase().substring(partitionArray[i].indexOf(" "),partitionArray[i].length()));
+        }
+        for(String busPartition:modifiedBusinessPartitionSet){
+            String onePartition="";
+            String[] eachPartitionValue = busPartition.split("/");
+            StringBuffer totalRow=new StringBuffer("");
+            for(int i=0; i<eachPartitionValue.length;i++){
+                String partitionDataType=partitionDataTypeMap.get(eachPartitionValue[i].split("=")[0].toUpperCase());
+                String colAndPartitionValue[]=eachPartitionValue[i].split("=");
+                if(partitionDataType.trim().contains("STRING")||partitionDataType.trim().contains("CHAR")||partitionDataType.trim().contains("DATE")||partitionDataType.trim().contains("TIME")) {
+                    colAndPartitionValue[1]="'"+colAndPartitionValue[1]+"'";
+                }
+                onePartition=colAndPartitionValue[0]+"="+colAndPartitionValue[1] +" AND ";
+                totalRow.append(onePartition);
+            }
+            String trimmedTotalRow=totalRow.substring(0,totalRow.lastIndexOf(" AND "));
+            filterCondition.append(trimmedTotalRow).append(" OR ");
+        }
+
+        return(filterCondition.toString().isEmpty()?"'a'='a'":filterCondition.substring(0,filterCondition.lastIndexOf(" OR ")));
+    }
+
+    private String removeDataTypesFromColumnList(String commaSeparatedColumns){
+        StringBuffer columnListWithoutDataTypes= new StringBuffer("");
+        String[] columns = commaSeparatedColumns.split(",");
+        for(int i=0;i<columns.length;i++){
+            columnListWithoutDataTypes.append(columns[i].substring(0,columns[i].indexOf(" "))).append(",");
+        }
+        return columnListWithoutDataTypes.toString().substring(0,columnListWithoutDataTypes.length()-1);
+    }
+
+    private String getTableLocation(String sourceDb,String hiveConnection,String sourceStgTable) throws Exception{
+        Connection conn = getHiveJDBCConnection(sourceDb,hiveConnection);
+        Statement st = conn.createStatement();
+        ResultSet rs = st.executeQuery("desc formatted "+sourceStgTable);
+        String tableLocation=new String();
+        while (rs.next()) {
+            if(rs.getString(1).trim().equals("Location:")) tableLocation=rs.getString(2);
+        }
+        return tableLocation;
+    }
+
+    private MigrationPreprocessorInfo setOozieUtilProperties(String sourceDb,String destDb,String sourceStgtable,String table, String sourceJobTrackerAddress, String sourceNameNodeAddress, String destNameNodeAddress, String srcHiveConnection,String processId, String instanceExecId, MigrationPreprocessorInfo migrationPreprocessorInfo){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 11 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00107)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "srcHiveConnection". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)